src/linear_algebra/basis.lean

```diff
@@ -751,6 +751,82 @@ end
 
 end span
 
+lemma smul_group_linear_independent
+  {G : Type*} [hG : group G] [distrib_mul_action G R] [distrib_mul_action G M]
+  [is_scalar_tower G R M] [smul_comm_class G R M] {v : basis ι R M} {w : ι → G} :
+  linear_independent R (w • v) :=
+begin
+  have hw₁' := v.linear_independent,
+  rw linear_independent_iff'' at hw₁' ⊢,
+  intros s g hgs hsum i,
+  refine (smul_eq_zero_iff_eq (w i)).1 _,
+  refine hw₁' s (λ i, w i • g i) (λ i hi, _) _ i,
+  { dsimp only,
+    exact (hgs i hi).symm ▸ smul_zero _ },
+  { rw [← hsum, finset.sum_congr rfl _],
+    intros, erw [pi.smul_apply, smul_assoc, smul_comm] },
+end
+
+lemma smul_group_span_eq_top
+  {G : Type*} [group G] [distrib_mul_action G R] [distrib_mul_action G M]
+  [is_scalar_tower G R M] {v : basis ι R M} {w : ι → G} :
+  submodule.span R (set.range (w • v)) = ⊤ :=
+begin
+  have hw₁' := v.span_eq,
+  rw eq_top_iff,
+  intros j hj,
+  rw ← hw₁' at hj,
+  rw submodule.mem_span at hj ⊢,
+  refine λ p hp, hj p (λ u hu, _),
+  obtain ⟨i, rfl⟩ := hu,
+  have : ((w i)⁻¹ • 1 : R) • w i • v i ∈ p := p.smul_mem ((w i)⁻¹ • 1 : R) (hp ⟨i, rfl⟩),
+  rwa [smul_one_smul, inv_smul_smul] at this,
+end
+
+/-- Given a basis `v` and a map `w` such that for all `i`, `w i` are elements of a group,
+`smul_group` provides the basis corresponding to `w • v`. -/
+def smul_group {G : Type*} [group G] [distrib_mul_action G R] [distrib_mul_action G M]
+  [is_scalar_tower G R M] [smul_comm_class G R M] (v : basis ι R M) (w : ι → G) :
+  basis ι R M :=
+@basis.mk ι R M (w • v) _ _ _ smul_group_linear_independent smul_group_span_eq_top
+
+lemma smul_group_apply {G : Type*} [group G] [distrib_mul_action G R] [distrib_mul_action G M]
+  [is_scalar_tower G R M] [smul_comm_class G R M] {v : basis ι R M} {w : ι → G} (i : ι) :
+  v.smul_group w i = (w • v : ι → M) i :=
+mk_apply smul_group_linear_independent smul_group_span_eq_top i
+
+-- This lemma cannot be proved with `smul_group_linear_independent` since the action of
+-- `units R` on `R` is not commutative.
```

Comment:
Are you sure this comment is still true?

Reply:
I tried but failed to prove `smul_group_linear_independent` without the `smul_comm_class` instance. Though I don't know if the problem is just with my method

Reply:
I thought I provided the instance in the other PR; what instance specifically is missing?

Reply:
I don't think this instance is true: `smul_comm_class (units R) R M`

Reply:
I suspect `smul_group_linear_independent` can be proved without the `smul_comm_class G R M` instance though I couldn't figure out how to do it :/

Reply:
I'll have a go tomorrow to see if I missed something

Reply:
Can you just adapt the units proof to work for an arbitrary group?

Reply:
The proof of the units version uses the smul action of  `R` on `units R` but we don't have the corresponding action of `R` on `G` in the group version